Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp

```diff
@@ -3434,94 +3447,126 @@ static bool consumeDeprecatedGradientColorStop(CSSParserTokenRange& range, CSSGr
     return stop.color && args.atEnd();
 }
 
+static std::optional<CSSGradientColorStopList> consumeDeprecatedGradientColorStops(CSSParserTokenRange& args, const CSSParserContext& context)
+{
+    CSSGradientColorStopList stops;
+    while (consumeCommaIncludingWhitespace(args)) {
+        CSSGradientColorStop stop;
+        if (!consumeDeprecatedGradientColorStop(args, stop, context))
+            return std::nullopt;
+        stops.append(WTFMove(stop));
+    }
+    stops.shrinkToFit();
+
+    return { WTFMove(stops) };
+}
+
 static AlphaPremultiplication gradientAlphaPremultiplication(const CSSParserContext& context)
 {
     return context.gradientPremultipliedAlphaInterpolationEnabled ? AlphaPremultiplication::Premultiplied : AlphaPremultiplication::Unpremultiplied;
 }
 
-static RefPtr<CSSValue> consumeDeprecatedGradient(CSSParserTokenRange& args, const CSSParserContext& context)
+static RefPtr<CSSValue> consumeDeprecatedLinearGradient(CSSParserTokenRange& args, const CSSParserContext& context)
```

Comment:
I am confused about all these consume functions, and I figured you might know the answer.

Which ones are allowed to consume some partial parsed content when the fail? Or is the contract that they must only move the range forward if the succeed?

Also, when should we name this `args` and when should we name it `range`?

Reply:
Also, hahaha that you are reviewing this. I was trying to be silly :). By the way, your PR, https://github.com/WebKit/WebKit/pull/5986, is looking great! I started putting to words some ideas for CSS properties in the new wiki - https://github.com/WebKit/WebKit/wiki/DRAFT:-Proposal-for-Generated-&-Modernized-CSS-Properties-and-Values